erpnext/buying/doctype/supplier_scorecard/supplier_scorecard.py

```diff
@@ -0,0 +1,237 @@
+		if make_all_scorecards(sc.name) > 0:
+			# Save the scorecard to update the score and standings
+			sc.save()
+
+
+@frappe.whitelist()
+def make_all_scorecards(docname):
+
+	sc = frappe.get_doc('Supplier Scorecard', docname)
+	supplier = frappe.get_doc('Supplier',sc.supplier)
+
+	start_date = getdate(supplier.creation)
+	end_date = get_scorecard_date(sc.period, start_date)
+	todays = getdate(nowdate())
+
+	scp_count = 0
+	first_start_date = todays
+	last_end_date = todays
+
+	while (start_date < todays) and (end_date <= todays):
+		# check to make sure there is no scorecard period already created
+		scorecards = frappe.db.sql("""
+			SELECT
+				scp.name
+			FROM
+				`tabSupplier Scorecard Period` scp
+			WHERE
+				scp.scorecard = %(sc)s
+				AND (
+					(scp.start_date > %(end_date)s
+					AND scp.end_date < %(start_date)s)
+				OR
+					(scp.start_date < %(end_date)s
+					AND scp.end_date > %(start_date)s))
+			ORDER BY
+				scp.end_date DESC""",
+				{"sc": docname,"start_date": start_date,"end_date": end_date, "supplier": supplier}, as_dict=1)
+		if len(scorecards) == 0:
+			period_card = make_supplier_scorecard(docname, None)
+			period_card.start_date = start_date
+			period_card.end_date = end_date
+			period_card.save()
+			scp_count = scp_count + 1
+			if start_date < first_start_date:
+				first_start_date = start_date
+			last_end_date = end_date
+
+		start_date = getdate(add_days(end_date,1))
+		end_date = get_scorecard_date(sc.period, start_date)
+	if scp_count > 0:
+		frappe.msgprint(_("Created {0} scorecards for ".format(scp_count) + sc.supplier + " between: ") + str(first_start_date) + " - " + str(last_end_date))
+	return scp_count
+
+def get_scorecard_date(period, start_date):
+	if period == 'Per Week':
+		end_date = getdate(add_days(start_date,7))
+	elif period == 'Per Month':
+		end_date = get_last_day(start_date)
+	elif period == 'Per Year':
+		end_date = add_days(add_years(start_date,1), -1)
+	return end_date
+
+def make_install_records():
+	install_docs = [
+		{"param_name":"total_accepted_items","name":"Total Accepted Items","doctype":"Supplier Scorecard Variable","variable_label":"Total Accepted Items","path":"get_total_accepted_items"},
+		{"param_name":"total_accepted_amount","name":"Total Accepted Amount","doctype":"Supplier Scorecard Variable","variable_label":"Total Accepted Amount","path":"get_total_accepted_amount"},
+		{"param_name":"total_rejected_items","name":"Total Rejected Items","doctype":"Supplier Scorecard Variable","variable_label":"Total Rejected Items","path":"get_total_rejected_items"},
+		{"param_name":"total_rejected_amount","name":"Total Rejected Amount","doctype":"Supplier Scorecard Variable","variable_label":"Total Rejected Amount","path":"get_total_rejected_amount"},
+		{"param_name":"total_received_items","name":"Total Received Items","doctype":"Supplier Scorecard Variable","variable_label":"Total Received Items","path":"get_total_received_items"},
```

Comment:
As naming is based on `variable_label`, you don't need to pass `name` explicitely.

Also, you can use doctype in common, to cleanup the data, it's really unreadable. Also break the lines after 80 chars.